pkg/auth/auth.go

```diff
@@ -153,3 +161,189 @@ func (c *Client) RevokeToken(ctx context.Context, req *RevokeTokenRequest) error
+type Helper struct {
+	Username  string
+	Password  string
+	TokenData *Tokenstore
+	apiMutex  sync.Mutex
+	API       API
+
+	fileMutex sync.Mutex
+}
+
+// NewHelper creates a new Helper instance
+func NewHelper(api API) (*Helper, error) {
+	err := godotenv.Load()
+	if err != nil {
+		return nil, fmt.Errorf("error loading .env file: %v", err)
+	}
+
+	username := os.Getenv("WME_USERNAME")
+	password := os.Getenv("WME_PASSWORD")
+	if username == "" || password == "" {
+		return nil, errors.New("username or password not set in .env file")
+	}
+
+	return &Helper{
+		Username: username,
+		Password: password,
+		API:      api,
+	}, nil
+}
+
+// GetAccessToken manages the token state and returns a valid access token
+func (h *Helper) GetAccessToken() (string, error) {
+	h.apiMutex.Lock()
+	defer h.apiMutex.Unlock()
+
+	// Check if the tokenstore file exists
+	if _, err := os.Stat(tokenStoreFile); os.IsNotExist(err) {
+		// File does not exist, create and login
+		return h.loginAndStoreTokens(tokenStoreFile)
+	}
+
+	// File exists, read and unmarshal token data
+	data, err := os.ReadFile(tokenStoreFile)
+	if err != nil {
+		return "", err
+	}
+
+	var tokenStore Tokenstore
+	if err := json.Unmarshal(data, &tokenStore); err != nil {
+		return "", err
+	}
+
+	// Check if the access token is still valid
+	if time.Since(tokenStore.AccessTokenGeneratedAt) < 24*time.Hour {
+		return tokenStore.AccessToken, nil
+	}
+
+	// Check if the refresh token is still valid
+	if time.Since(tokenStore.RefreshTokenGeneratedAt) < 30*24*time.Hour {
+		return h.refreshAndStoreTokens(tokenStoreFile, &tokenStore)
+	}
+
+	// Both tokens are expired, login again
+	return h.loginAndStoreTokens(tokenStoreFile)
+}
+
+// loginAndStoreTokens logs in and stores the tokens
+func (h *Helper) loginAndStoreTokens(tokenStoreFile string) (string, error) {
+	loginRequest := &LoginRequest{
+		Username: h.Username,
+		Password: h.Password,
+	}
+
+	loginResponse, err := h.API.Login(context.Background(), loginRequest)
+	if err != nil {
+		return "", err
+	}
+
+	tokenStore := &Tokenstore{
+		AccessToken:             loginResponse.AccessToken,
+		AccessTokenGeneratedAt:  time.Now(),
+		RefreshToken:            loginResponse.RefreshToken,
+		RefreshTokenGeneratedAt: time.Now(),
+	}
+
+	if err := h.storeTokens(tokenStoreFile, tokenStore); err != nil {
+		return "", err
+	}
+
+	return tokenStore.AccessToken, nil
+}
+
+// refreshAndStoreTokens refreshes the access token and stores the new tokens
+func (h *Helper) refreshAndStoreTokens(tokenStoreFile string, tokenStore *Tokenstore) (string, error) {
+	refreshRequest := &RefreshTokenRequest{
+		Username:     h.Username,
+		RefreshToken: tokenStore.RefreshToken,
+	}
+
+	var refreshResponse *RefreshTokenResponse
+	var err error
+
+	// Check if the refresh token is less than 30 days old
+	if time.Since(tokenStore.RefreshTokenGeneratedAt) < 30*24*time.Hour {
+		refreshResponse, err = h.API.RefreshToken(context.Background(), refreshRequest)
+	}
```

Comment:
We can directly call `h.API.RefreshToken()` without checking this condition, since we only call `refreshAndStoreTokens()` when this condition is true.

Reply:
Remove condition, thanks :)